src/goto-symex/goto_symex_state.cpp

```diff
@@ -495,12 +493,12 @@ void goto_symex_statet::rename(
   }
   else if(expr.id()==ID_address_of)
   {
+    DATA_INVARIANT(
+      expr.type().id() == ID_pointer,
+      "type of expression required to be pointer");
```

Comment:
I think this should just be removed, as should the `DATA_INVARIANT` below. We don't even care whether the type of the expression is a pointer, and the subsequent invariant should be replaced by a use of `to_address_of_expr`